app/src/controllers/object.js

```diff
@@ -297,20 +298,32 @@ const controller = {
           bucketId: bucketId
         });
 
+        // sync existing file versions and get COMS objectId
+        const objectId =
+          await syncService.syncJob(joinPath(bucketKey, req.currentUpload.filename), bucketId, true, userId);
+
         // Hard short circuit skip file as the object already exists on bucket
-        throw new Problem(409, {
+        throw new Problem(409, 'Bucket already contains object', req.originalUrl, {
           detail: 'Bucket already contains object',
-          instance: req.originalUrl
+          existingObjectId: objectId,
         });
+
       } catch (err) {
         if (err instanceof Problem) throw err; // Rethrow Problem type errors
 
         // Object is soft deleted from the bucket
+        // TODO: add coms-id to error message if exists
         if (err.$response?.headers['x-amz-delete-marker']) {
-          throw new Problem(409, {
-            detail: 'Bucket already contains object',
-            instance: req.originalUrl
+
+          // sync existing file versions and get COMS objectId
+          const objectId =
+            await syncService.syncJob(joinPath(bucketKey, req.currentUpload.filename), bucketId, true, userId);
+
+          throw new Problem(409, 'Bucket already contains a soft-deleted version of this object', req.originalUrl, {
+            detail: 'Bucket already contains a soft-deleted version of this object',
```

Comment:
IMO I'd suggest in the `detail` to upload a new version to the existing object instead, but I don't think our other `detail` messages (presumably by design) have this kind of `{error} + {suggestion}` format. Just a thought 🤷